src/yaml/mapping.cr

```diff
@@ -181,6 +181,14 @@ module YAML
             {% end %}
           end
         {% end %}
+        {% if extra %}
+          _{{extra.id}} = @{{extra.id}}
```

Comment:
In this implementation extra is nil unless the user has set it (and passed it in as an argument to the `mapping(...)` macro. Because the macro checks for existence of extra before emitting code,  `@{{extra.id}}` won't exist at all if the user is not using extras. If extra does exist then `@{{extra.id}}` will always get initialized as a `Hash(String, ::YAML::Any)`

If I'm understanding this correctly, the only problem with trying to call `@{{extra.id}}.to_yaml(...` would be if `@{{extra.id}}` is nil at the time of executing this block and that's impossible. I hope I'm understanding correctly 😄